src/azure-cli/azure/cli/command_modules/cosmosdb/tests/latest/test_cosmosdb_commands.py

```diff
@@ -1410,10 +1412,77 @@ def test_cosmosdb_managed_service_identity(self, resource_group, key_vault):
+        assert identity_output["type"] == "UserAssigned"
+        assert len(identity_output["user_identities"]) == 2
+
+        identity_output = self.cmd('az cosmosdb identity remove {0} -n {acc} -g {rg}', user_identity2["id"]).get_output_in_json()
+        assert identity_output["type"] == "UserAssigned"
+        assert identity_output["user_identities"][0][0] == user_identity1["id"]
+        assert len(identity_output["user_identities"]) == 1
+
+        identity_output = self.cmd('az cosmosdb identity assign -n {acc} -g {rg} --identities {0}', user_identity2["id"]).get_output_in_json()
+        assert identity_output["type"] == "UserAssigned"
+        assert identity_output["user_identities"][1][0] == user_identity2["id"]
+        assert len(identity_output["user_identities"]) == 2
+
+        identity_output = self.cmd('az cosmosdb identity remove -n {acc} -g {rg} --identities {0} {1}', user_identity1["id"], user_identity2["id"]).get_output_in_json()
+        assert identity_output["type"] == "None"
+
+        identity_output = self.cmd('az cosmosdb identity assign -n {acc} -g {rg} --identities {0} {1} [system]', user_identity1["id"], user_identity2["id"]).get_output_in_json()
+        assert identity_output["type"] == "SystemAssigned,UserAssigned"
+        assert len(identity_output["user_identities"]) == 2
+        
+        identity_output = self.cmd('az cosmosdb identity remove -n {acc} -g {rg} --identities {0}', user_identity2["id"]).get_output_in_json()
+        assert identity_output["type"] == "SystemAssigned,UserAssigned"
+        assert len(identity_output["user_identities"]) == 1
+        
+        identity_output = self.cmd('az cosmosdb identity remove -n {acc} -g {rg} --identities {0}', user_identity1["id"]).get_output_in_json()
+        assert identity_output["type"] == "SystemAssigned"
+        
+        identity_output = self.cmd('az cosmosdb identity assign -n {acc} -g {rg} --identities {0} {1} [system]', user_identity1["id"], user_identity2["id"]).get_output_in_json()
+        assert identity_output["type"] == "SystemAssigned,UserAssigned"
+        assert len(identity_output["user_identities"]) == 2
+        identity_output = self.cmd('az cosmosdb identity remove -n {acc} -g {rg} --identities {0} {1} [system]', user_identity1["id"], user_identity2["id"]).get_output_in_json()
+        assert identity_output["type"] == "None"
+
+        # Default identity tests
+        identity_output = self.cmd('az cosmosdb identity assign -n {acc} -g {rg}').get_output_in_json()
+        identity_output = self.cmd('az cosmosdb update -n {acc} -g {rg} --default-identity SystemAssignedIdentity').get_output_in_json()
+        assert identity_output["defaultIdentity"] == 'SystemAssignedIdentity'
+
+        identity_output = self.cmd('az cosmosdb update -n {acc} -g {rg} --default-identity UserAssignedIdentity={0}', user_identity1["id"]).get_output_in_json()
```

Comment:
Will this work? Because you have not setup the KeyVault access policy to let the user_identity1 to have access to the KeyVault yet. This call will get a BadRequest.